frontend/src/app/components/Leaderboards/Leaderboards.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react'
+import { connect } from 'react-redux'
+
+import { ListCon, FiltersBlock, FiltersSpan } from '../../../styles/blocks'
+import { LeaderboardsRow } from './LeaderboardsRow'
+import { getTopUsers } from '../../modules/users/users-selectors'
+import { LeaderboardsActions } from '../../modules/leaderboards/leaderboards-actions'
+import * as Filters from '../../const/leaderboards-filters'
+
+const LeaderboardsComponent = (
+  { topUsers, filters, maxItems, updateCriteria, updateOrder, updateTimespan }
+) =>
+  (
+  <>
+    <FiltersBlock>
+      {Object.entries(Filters.criteriaTypes).map(([key, value]) =>
+        <FiltersSpan key={key} className={value === filters.criteria ? 'active-filter' : ''}
+          onClick={() => updateCriteria(value)}>{key}</FiltersSpan>,
+      )}
+
+      {Object.entries(Filters.orderTypes).map(([key, value]) =>
+        <FiltersSpan key={key} className={value === filters.order ? 'active-filter' : ''}
+          onClick={() => updateOrder(value)}>{value}</FiltersSpan>,
+      )}
+
+      {Object.entries(Filters.timespanTypes).map(([key, value]) =>
+        <FiltersSpan key={key} className={value === filters.timespan ? 'active-filter' : ''}
+          onClick={() => updateTimespan(value)}>{key.split(/(?=[A-Z0-9])/).join(' ')}
+        </FiltersSpan>,
+      )}
+    </FiltersBlock>
+
+    <ListCon className="topPlayers" ascending={filters.order === Filters.orderTypes.ASC}>
+      {topUsers.slice(0, maxItems).map((user, index) =>
+        <LeaderboardsRow key={user.id} user={user} 
+          position={filters.order === Filters.orderTypes.ASC ? topUsers.length - index : index + 1}
+          points={getUserPoints(user, filters.criteria)} />,
+      )}
+    </ListCon>
+  </>
+  )
+
+const getUserPoints = (user, criteria) => {
+  switch (criteria) {
+    case Filters.criteriaTypes.Wins:
+      return user.totalWins
+    case Filters.criteriaTypes.Ratio:
+      return (user.winRatio* 100).toFixed(2) + '%'
+    case Filters.criteriaTypes.Streak:
+      return user.longestStreak
+    case Filters.criteriaTypes.Matches:
+      return user.totalMatches
+    default:
+      return user.rating
+  }
+}
+
+const mapStateToProps = state => ({
+  topUsers: getTopUsers(state),
+  filters: state.filters,
```

Comment:
if you notice, the only reason you use filters here is because of filters.criteria. In the ideal design the component would be dumb and not even have that thing in the props (so ideally only provide the topUsers according to the active criterion)

Reply:
topUsers provided by getTopUsers are already sorted by filters. I pass the filters to props so I can use them inside component to highlight active filters and generate filters header with cycles.

Reply:
That makes sense, but it would still be better to generate the user objects with the values in the selector and let the component just be dumb and render these objects IMHO

Reply:
Made the component dumb [here](https://github.com/salsita/foosball-rating/pull/92/commits/66516b36eac73b2ce0f63c758689cac523dc09f9#diff-f82bc9601bbd1ef2caeef4b93a143ab9) and increased selectors power [here](https://github.com/salsita/foosball-rating/pull/92/commits/66516b36eac73b2ce0f63c758689cac523dc09f9#diff-6bb144976a04c9e3de620e0c0828e32e)